### ISSUE TITLE
Handle conditional case statements

### ISSUE THREAD

**Issue by Elscha** [KernelHaven/srcMLExtractor#4]
Parsing
```C
switch (no) {
	case arg_AX:
		rv = &regs->ax;
		break;
	case arg_BX:
		rv = &regs->bx;
		break;
	case arg_CX:
		rv = &regs->cx;
		break;
	case arg_DX:
		rv = &regs->dx;
		break;
	case arg_SP:
		rv = &regs->sp;
		break;
	case arg_BP:
		rv = &regs->bp;
		break;
	case arg_SI:
		rv = &regs->si;
		break;
	case arg_DI:
		rv = &regs->di;
		break;
#ifdef __amd64__
	case arg_R8:
		rv = &regs->r8;
		break;
	case arg_R9:
		rv = &regs->r9;
		break;
	case arg_R10:
		rv = &regs->r10;
		break;
	case arg_R11:
		rv = &regs->r11;
		break;
	case arg_R12:
		rv = &regs->r12;
		break;
	case arg_R13:
		rv = &regs->r13;
		break;
	case arg_R14:
		rv = &regs->r14;
		break;
	case arg_R15:
		rv = &regs->r15;
		break;
#endif
	default:
		printk(KERN_ERR "mmiotrace: Error reg no# %d\n", no);
	}

	return rv;
}
```

Will result in:
```Java
java.lang.IllegalArgumentException: Illegal element net.ssehub.kernel_haven.srcml.transformation.CodeUnit while translating C:\Repos\GIT\KernelHaven\srcMLExtractor\real\Linux4.15\pf_in.c
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convert(TranslationUnitToAstConverter.java:101)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convertPreprocessorBlock(TranslationUnitToAstConverter.java:413)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convert(TranslationUnitToAstConverter.java:96)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convertTranslationUnit(TranslationUnitToAstConverter.java:278)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convert(TranslationUnitToAstConverter.java:94)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.createTypeDef(TranslationUnitToAstConverter.java:388)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convertTranslationUnit(TranslationUnitToAstConverter.java:223)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convert(TranslationUnitToAstConverter.java:94)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convertTranslationUnit(TranslationUnitToAstConverter.java:252)
	at net.ssehub.kernel_haven.srcml.transformation.TranslationUnitToAstConverter.convert(TranslationUnitToAstConverter.java:94)
	at net.ssehub.kernel_haven.srcml.transformation.XmlToSyntaxElementConverter.getAst(XmlToSyntaxElementConverter.java:199)
	at net.ssehub.kernel_haven.srcml.xml.AbstractAstConverter.getResult(AbstractAstConverter.java:61)
	at net.ssehub.kernel_haven.srcml.xml.XmlToAstConverter.parseToAst(XmlToAstConverter.java:56)
	at net.ssehub.kernel_haven.srcml.SrcMLExtractor.runOnFile(SrcMLExtractor.java:110)
	at net.ssehub.kernel_haven.srcml.AbstractSrcMLExtractorTest.loadFile(AbstractSrcMLExtractorTest.java:107)
	at net.ssehub.kernel_haven.srcml.RobustnessTests.testAllFiles(RobustnessTests.java:37)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
```

**Comment by adam-sse** [comment]
Cannot reproduce.

**Comment by adam-sse** [comment]
The actual problem was an enum with an `#ifdef`:
```C
enum {
	arg_DI = 7,
#ifdef __amd64__
	arg_R8  = 8,
#endif
};
```
Fixed by 80845c9c964b92e795afe90237fc3401df7514d2
